kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CoordinatedCommitsSuite.scala

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.internal.coordinatedcommits
+
+import io.delta.kernel.defaults.DeltaTableWriteSuiteBase
+import io.delta.kernel.defaults.internal.logstore.LogStoreProvider
+import io.delta.kernel.defaults.utils.TestRow
+import io.delta.kernel.internal.TableConfig._
+import io.delta.kernel.Table
+import io.delta.kernel.data.Row
+import io.delta.kernel.defaults.engine.DefaultCommitCoordinatorClientHandler
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.internal.{SnapshotImpl, TableConfig}
+import io.delta.kernel.internal.actions.{CommitInfo, Metadata, Protocol, SingleAction}
+import io.delta.kernel.internal.actions.SingleAction.{createMetadataSingleAction, FULL_SCHEMA}
+import io.delta.kernel.internal.util.{CoordinatedCommitsUtils, FileNames}
+import io.delta.kernel.internal.util.Preconditions.checkArgument
+import io.delta.kernel.internal.util.Utils.{singletonCloseableIterator, toCloseableIterator}
+import io.delta.kernel.utils.{CloseableIterator, FileStatus}
+import io.delta.storage.commit.{CommitCoordinatorClient, CommitResponse, GetCommitsResponse, InMemoryCommitCoordinator, UpdatedActions}
+import io.delta.storage.commit.actions.{AbstractMetadata, AbstractProtocol}
+import io.delta.storage.LogStore
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+
+import java.{lang, util}
+import java.util.{Collections, Optional}
+import scala.collection.convert.ImplicitConversions.`iterator asScala`
+import scala.collection.JavaConverters._
+
+class CoordinatedCommitsSuite extends DeltaTableWriteSuiteBase
+  with CoordinatedCommitsTestUtils {
+
+  test("cold snapshot initialization") {
+    InMemoryCommitCoordinatorBuilder.clearInMemoryInstances()
+    withTempDirAndEngine ({ (tablePath, engine) =>
+      val handler =
+        engine.getCommitCoordinatorClientHandler("tracking-in-memory", Collections.emptyMap())
+      val logPath = new Path("file:" + tablePath, "_delta_log")
+      val tableSpark = Table.forPath(engine, tablePath)
+
+      spark.range(0, 10).write.format("delta").mode("overwrite").save(tablePath) // version 0
```

Comment:
Can you move the setup code to a common function? And please add some comments to describe what is happening here.